internal/provider/datasources/block.go

```diff
@@ -0,0 +1,198 @@
+package datasources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+	"github.com/prefecthq/terraform-provider-prefect/internal/provider/customtypes"
+	"github.com/prefecthq/terraform-provider-prefect/internal/provider/helpers"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &blockDataSource{}
+	_ datasource.DataSourceWithConfigure = &blockDataSource{}
+)
+
+// blockDataSource is the data source implementation.
+type blockDataSource struct {
+	client api.PrefectClient
+}
+
+// BlockDataSourceModel defines the Terraform data source model.
+type BlockDataSourceModel struct {
+	ID          customtypes.UUIDValue      `tfsdk:"id"`
+	Created     customtypes.TimestampValue `tfsdk:"created"`
+	Updated     customtypes.TimestampValue `tfsdk:"updated"`
+	AccountID   customtypes.UUIDValue      `tfsdk:"account_id"`
+	WorkspaceID customtypes.UUIDValue      `tfsdk:"workspace_id"`
+
+	Name          types.String         `tfsdk:"name"`
+	Data          jsontypes.Normalized `tfsdk:"data"`
+	BlockTypeName types.String         `tfsdk:"block_type_name"`
```

Comment:
@parkedwards - could you do a quick sanity check here? Want to make sure I didn't miss anything - could just be inconsistency in the API field names.

Reply:
yeah - so the choice i made (and it's open for debate) was that - since `prefect_block` is itself kind of a unification of multiple API resources, i figured we could do the same and just obfuscate the notion of a `block_type` and just call it a `type` on the resource, since we're already abstracting those underlying APIs to the user

and then the `_slug` aspect was to differentiate a type's `name` and a `slug` - in general, they are similar, but the `slug` is the hyphenated string, whereas i _**think**_ the name is cap-cased and has spaces?

Reply:
```
        "name": "Azure Container Instance Job",
        "slug": "azure-container-instance-job",
```